src/scrap.py

```diff
@@ -0,0 +1,254 @@
+
+    div_post = parse_content.find_all(
+        "div",
+        class_="base-card relative w-full hover:no-underline focus:no-underline base-card--link base-search-card base-search-card--link job-search-card",
+    )
+
+    linked_in_jobs = []
+
+    for elem in div_post:
+        job = {}
+        job["_company"] = elem.find(
+            "h4", class_="base-search-card__subtitle"
+        ).text.strip()
+        job["_title"] = elem.find("h3", class_="base-search-card__title").text.strip()
+        job["_location"] = elem.find(
+            "span", class_="job-search-card__location"
+        ).text.strip()
+        job["_link"] = elem.find("a", class_="base-card__full-link")["href"]
+
+        linked_in_jobs.append(job)
+
+    return linked_in_jobs
+
+
+# -------------------FOR HELPER FUNCTIONS ----------------------------
+
+# List opportunities shows all the opportunities that have NOT been posted yet
+
+
+def list_opportunities() -> List[object]:
+    connection = sqlite3.connect(job_db)
+    cursor = connection.cursor()
+
+    cursor.execute("SELECT * FROM jobs")
+    rows = cursor.fetchall()
+
+    # Printing out for debugging purposes
+    for row in rows:
+        _company, _title, _location, _link, _processed = zip(*row)
+
+        if not _processed:
+            print("Company:", _company)
+            print("Title:", _title)
+            print("Location:", _location)
+            print("Link:", _link)
+            print("Processed:", _processed)
+            print(" ")
+
+    connection.close()
+
+    return rows
+    # TO-DO: RETURN THE JOBS THAT HAVENT BEEN POSTED
+
+
+# -------------------FOR DISCORD BOT----------------------------
+
+
+intents = discord.Intents.default()
+intents.message_content = True
+client = discord.Client(intents=intents)
+discord_token = os.getenv("DISCORD_TOKEN")
+
+# TO DO - Ask Ethan about the web hook thingy kabob
+# webhook = /HIDDEN/
+
+
+async def execute_opportunities_webhook():
+    channel_id = os.getenv("DISCORD_CHANNEL_ID")
+
+    channel = client.get_channel(channel_id)
+    await channel.send("testing...")
+
+    data_results = list_opportunities()
+
+    for element in data_results:
+        company, title, location, link, status = zip(*element)
+
+        # Checking to see if the PROCESSED status is 0, and if so, means false.
+        if status == 0:
+            await channel.send(
+                f"NEW JOB! {company} is hiring for {title} at {location}!!! Link: {link}"
+            )
+
+        connection = sqlite3.connect(job_db)
+        cursor = connection.cursor()
+
+        # Updating the PROCESSED status to 1, which means its already been posted.
+        cursor.execute(
+            "UPDATE jobs SET status = ? WHERE company = ? AND title = ?",
+            (1, company, title),
+        )
+
+        connection.commit()
+        connection.close()
+
+
+@client.event
+async def on_ready():
+    await execute_opportunities_webhook()
+
+
+client.run(discord_token)
```

Comment:
To kick off the program, I recommend following the `if __name__ == "main"` Python convention.
```suggestion
def main():
    # Your main code logic goes here
    await execute_opportunities_webhook()

if __name__ == "__main__":
    main()
```